### PR TITLE
Content for OAuth2/OIDC support for Kafka REST proxy

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
# What changed, and why it matters

Added content for OAuth2/OIDC support for Kafka REST proxy - [DOC-592](https://aiven.atlassian.net/browse/DOC-592)

Note: I will remove the [Enable Apache Kafka® REST proxy authorization topic](https://docs.aiven.io/docs/products/kafka/karapace/howto/enable-kafka-rest-proxy-authorization), as I have included this content within this new topic "Enable OAuth2/OIDC support for Apache Kafka® REST proxy".

[DOC-592]: https://aiven.atlassian.net/browse/DOC-592?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+OAuth2/OIDC token handling
+---------------------------
+
+Karapace processes the JSON Web Token (JWT) obtained from the Authorization HTTP header, specifically when employing the Bearer authentication scheme. This allows OAuth2/OIDC credentials to be supplied directly to the REST proxy, which uses the provided token to authorize requests to Apache Kafka. When a Bearer token is presented, Kafka clients configured by Karapace use the SASL OAUTHBEARER mechanism to send the JWT for validation.
+
+
+Authorization enforcement
+----------------------------
+
+The underlying Aiven for Apache Kafka® service uses the ``sub`` claim from the JWT as the username, checking it against configured Access Control Lists (ACLs) to authorize operations on Kafka resources.
```
Comment (matyaskuti) [inline docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst:15]: A small thing here, the default is indeed the `sub` claim, however on the Kafka side this is configurable via `kafka.sasl_oauthbearer_sub_claim_name`, as also mentioned in https://docs.aiven.io/docs/products/kafka/howto/enable-oidc#enable-oauth2-oidc-via-aiven-console

Comment (jjaakola-aiven) [reply docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst:15]: And currently the service user matching the value in the claim is needed, just the ACL entry is not enough.

Comment (harshini-rangaswamy) [reply docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst]: Thanks both for your feedback. I have made some changes to the text and would appreciate it if you could review it again.

**Review by matyaskuti — APPROVED** [review]
Looks good to me!
@jjaakola-aiven could you also check again?

File: docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+Authorization enforcement
+----------------------------
+
+In the underlying Aiven for Apache Kafka® service, the default mechanism for authorization, uses the ``sub`` claim from the JWT as the username. This username is then verified against the configured Access Control Lists (ACLs) to authorize user operations on Kafka resources.
+
+While the ``sub`` claim is the default identifier, this setting is configurable. You can specify a different JWT claim for authentication by adjusting the ``kafka.sasl_oauthbearer_sub_claim_name`` parameter. For more information on configuring this, see :ref:`console-authentication`.
+
+To authenticate and authorize a user in Aiven for Apache Kafka, the JWT claim value and corresponding service user must be explicitly present in the ACL. This means that the identity of the user making the request with the JWT claim value must match the service user in the system.
```
[the remark this answers is not in the document]
Comment (harshini-rangaswamy) [reply docs/products/kafka/karapace/howto/enable-oauth-oidc-kafka-rest-proxy.rst]: @jjaakola-aiven I've made the following text updates. Could you check them and confirm if they're alright?

To authenticate and authorize a user in Aiven for Apache Kafka, both a service user and an ACL entry describing the permissions are required. The JWT claim value used for authentication must explicitly match the service user in the system. Furthermore, this service user must be associated with an ACL entry that outlines their permissions, ensuring that the identity of the user making the request aligns with both the service user and the ACL entry.

**Review by jjaakola-aiven — APPROVED** [review]

**Pull request merged**